### PR TITLE
Boss_Fd2

### PR DESCRIPTION
**Pull request by petrie911**
Splitting this off from my other pull request. Matched and documented.

### CODE REVIEW

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 1]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+                    i1 = (s16)Math_Rand_ZeroFloat(8.9f);
+                    this->actor.posRot.pos.x = D_808D6124[i1].x;
+                    this->actor.posRot.pos.z = D_808D6124[i1].z;
+                    this->actionState = 0;
+                    this->timers[0] = 0xA;
+                } else {
+                    this->skelAnime1.animPlaybackSpeed = 1.0f;
+                    this->animationLength = SkelAnime_GetFrameCount(&D_0600C1D0.genericHeader);
+                    this->actionState = 2;
+                    Audio_PlayActorSound2(&this->actor, 0x38D1);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:347]: 0x38D1 corresponds to the constant NA_SE_EN_VALVAISA_ROAR

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 2]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+                    }
+                    bossFd->holeSplashTimer = 5;
+                }
+            }
+            break;
+        case 2:
+            Math_SmoothScaleMaxS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 3, 0x7D0);
+            if ((this->timers[0] == 1) && (this->actor.xzDistFromLink < 120.0f)) {
+                func_8002F6D4(globalCtx, &this->actor, 3.0f, this->actor.yawTowardsLink, 2.0f, 0x20);
+                Audio_PlayActorSound2(&player->actor, 0x83E);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:365]: 0x83E should be replaced with NA_SE_PL_BODY_HIT

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 3]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime1);
+    if (func_800A56C8(&this->skelAnime1, this->animationLength) != 0) {
+        BossFd2_SetupBurrow(this, globalCtx);
+    }
+    if ((25.0f <= this->skelAnime1.animCurrentFrame) && (this->skelAnime1.animCurrentFrame < 70.0f)) {
+        if (this->skelAnime1.animCurrentFrame == 25.0f) {
+            globalCtx->envCtx.unk_D8 = 0.0f;
+        }
+        Audio_PlayActorSound2(&this->actor, 0x30DB);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:489]: 0x30DB = NA_SE_EN_VALVAISA_FIRE - SFX_FLAG

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 4]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime1, &D_0600B7A4, -5.0f);
+    this->actionFunc = BossFd2_ClawSwipe;
+    this->animationLength = SkelAnime_GetFrameCount(&D_0600B7A4.genericHeader);
+}
+
+void BossFd2_ClawSwipe(BossFd2* this, GlobalContext* globalCtx) {
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime1);
+    if (func_800A56C8(&this->skelAnime1, 5.0f) != 0) {
+        Audio_PlayActorSound2(&this->actor, 0x38D1);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:569]: 0x38D1 = NA_SE_EN_VALVAISA_ROAR

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 5]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+        Audio_PlayActorSound2(&this->actor, 0x38D8);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:570]: 0x38D8 = NA_SE_EN_VALVAISA_SW_NAIL

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 6]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+    Vec3f sp78;
+    Vec3f sp6C;
+
+    this->disableAT = 1;
+    this->actor.flags |= 0x400;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime1);
+    switch (this->actionState) {
+        case 0:
+            if (func_800A56C8(&this->skelAnime1, 13.0f) != 0) {
+                Audio_PlayActorSound2(&this->actor, 0x38D3);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:599]: 0x38D3 = NA_SE_EN_VALVAISA_MAHI2

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 7]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+                    sp84.x = Math_Rand_CenteredFloat(8.0f);
+                    sp84.y = Math_Rand_ZeroFloat(1.0f);
+                    sp84.z = Math_Rand_CenteredFloat(8.0f);
+                    sp78.y = 0.5f;
+                    sp6C.x = Math_Rand_CenteredFloat(10.0f) + this->actor.posRot2.pos.x;
+                    sp6C.y = Math_Rand_CenteredFloat(10.0f) + this->actor.posRot2.pos.y;
+                    sp6C.z = Math_Rand_CenteredFloat(10.0f) + this->actor.posRot2.pos.z;
+                    BossFd2_SpawnDust(bossFd->particles, &sp6C, &sp84, &sp78, Math_Rand_ZeroFloat(100.0f) + 300.0f);
+                }
+                Audio_PlayActorSound2(&this->actor, 0x3927);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:613]: 0x3927 = NA_SE_EN_VALVAISA_LAND

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 8]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+            }
+            if (func_800A56C8(&this->skelAnime1, this->animationLength) != 0) {
+                SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime1, &D_0600A86C, -5.0f);
+                this->actionState = 1;
+                this->timers[0] = 60;
+            }
+            break;
+        case 1:
+            if ((this->varianceTimer & 0xF) == 0xF) {
+                Audio_PlayActorSound2(&this->actor, 0x38D4);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:623]: 0x38D4 = NA_SE_EN_VALVAISA_KNOCKOUT

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 9]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+    this->disableAT = 1;
+    if (this->actionState == 0) {
+        if (func_800A56C8(&this->skelAnime1, this->animationLength) != 0) {
+            SkelAnime_ChangeAnimDefaultStop(&this->skelAnime1, &D_060089DC);
+            this->animationLength = SkelAnime_GetFrameCount(&D_060089DC.genericHeader);
+            this->actionState = 1;
+        }
+    } else if (this->actionState == 1) {
+        if (func_800A56C8(&this->skelAnime1, 6.0f) != 0) {
+            Audio_PlayActorSound2(&this->actor, 0x38D6);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:652]: 0x38D6 = NA_SE_EN_VALVAISA_DAMAGE2

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 10]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+            } else {
+                sp7C = 0.25f;
+            }
+            if ((this->holeCounter == 1) || (this->holeCounter == 40)) {
+                this->screamTimer = 20;
+                if (this->holeCounter == 40) {
+                    func_800F8D04(0x38D9);
+                }
+
+                Audio_PlayActorSound2(&this->actor, 0x38D6);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:746]: NA_SE_EN_VALVAISA_DAMAGE2

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 11]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+            osSyncPrintf("\x1b[m");
+            osSyncPrintf("hp %d\n", bossFd->actor.colChkInfo.health);
+
+            if ((s8)bossFd->actor.colChkInfo.health <= 0) {
+                bossFd->actor.colChkInfo.health = 0;
+                BossFd2_SetupDeath(this, globalCtx);
+                this->damageFlashTimer = 10;
+                this->invulnerabilityTimer = 30000;
+                Audio_SetBGM(0x100100FF);
+                Audio_PlayActorSound2(&this->actor, 0x38D9);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:951]: NA_SE_EN_VALVAISA_DEAD

File: src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c [hunk 12]
```diff
@@ -21,69 +87,1189 @@ const ActorInit Boss_Fd2_InitVars = {
...
+                func_80032C7C(globalCtx, &this->actor);
+            } else if (temp_v0_4 != 0) {
+                BossFd2_SetupDamaged(this, globalCtx);
+                this->damageFlashTimer = 10;
+                this->invulnerabilityTimer = 100;
+                Audio_PlayActorSound2(&this->actor, 0x38D5);
```
Comment (NEstelami) [inline src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c:957]: NA_SE_EN_VALVAISA_DAMAGE1

**Comment by fig02** [comment]
merged into https://github.com/zeldaret/oot/pull/352

**Pull request closed without merging**